library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java

```diff
@@ -378,30 +378,50 @@
     private final ContentResolver contentResolver;
     private final DownloadsUriProvider downloadsUriProvider;
     private final SystemFacade systemFacade;
+    private final BatchRepository batchRepository;
 
     private Uri baseUri;
 
-    public DownloadManager(Context context, ContentResolver resolver) {
-        this(context, resolver, DownloadsUriProvider.getInstance(), new RealSystemFacade(context), false);
+    public static DownloadManager newInstance(Context context, ContentResolver contentResolver) {
```

Comment:
This is the only sensible way to inject the dependency and avoid several instances of `downloadsUriProvider` and `realSystemFacade`.

I prefer testability rather than to use a non well known pattern that is not that complex